src/DIRAC/FrameworkSystem/Client/BundleDeliveryClient.py

```diff
@@ -134,13 +140,12 @@ def syncCAs(self):
         if "X509_CERT_DIR" in os.environ:
             X509_CERT_DIR = os.environ["X509_CERT_DIR"]
             del os.environ["X509_CERT_DIR"]
+        if X509_CERT_DIR:
```

Comment:
In the previous `if` the entry is removed from the dictionary, and then it is brought back here right away? Was `X509_CERT_DIR` not saved temporarily so the entry is not in `os.environ` while the next lines are executed before the entry is put back? Just guessing, didn't look at what is inside those functions...

Using `mock.patch.dict` we can make this look slightly nicer
```python
import mock
with mock.patch.dict("os.environ"):
   os.environ.pop("X509_CERT_DIR", None)
   #...
   return self.syncDir(casLocation)
```

Reply:
Oh you are right. I revert it.